fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -59,112 +56,45 @@ fun NavigationBar(
                 .padding(horizontal = 16.dp),
             horizontalArrangement = Arrangement.SpaceBetween,
         ) {
-            actionItems.forEach {
-                when (it.type) {
-                    ItemType.STANDARD -> {
-                        IconButton(onClick = {}) {
+            buttons.forEach { button ->
+                when (button) {
+                    is NavBarButton.ActionButton -> {
+                        LongPressIconButton(
```

Comment:
what was the motivation for having this UI creation be so dynamic? Was Product or UX wanting the flexibility of reordering the buttons or something?

Just wanting to understand what led you here versus other approaches. From I can glean here, there's only realy two variations of the UI.

Reply:
On a high level, I was thinking about the nav bar in the similar manner to the toolbar. We should probably someday make a AC component that would be customize and not tied to a particular design/implementation.

On a more practical level, the current design has four states and the dynamic way looked more appropriate.
Home screen (with a middle search button)
Browser screen (with a middle home button)
Browser screen, private mode (with a burn button)
Custom tab (with 4 buttons and a switch to the browser button)

Reply:
Gotcha gotcha. Thanks for listing out the four UI variations. 

What are the advantages you're wanting to achieve from this dynamic approach? What I'm seeing is four use cases that could also easily be statically constructed in their own Composables, having lambdas for each button's click/long click. Is having the one `NavigationBar` Composable worth the trade-off of the setup complexity from `NavigationBarButtons`?

Is there already some desire to upstream this to AC and then be consumed in Focus?

I'm asking these exploratory questions to help me (and other readers) understand why this is a candidate for our best approach for this problem.

Reply:
Now that I think of it, part of the reason was I was just not very experienced with the `Composable` :) 

I tried tom implement the suggested [approach](https://github.com/mozilla-mobile/firefox-android/pull/5888), is it something you had in mind?

It ended up being a bit more verbose, especially because of the previews.

Reply:
Maybe we can do a recorded session with @MatthewTighe when I'm back from PTO to go over our current state of fragments + Compose + redux.  In the meantime, my [walkthrough of the debug drawer code](https://bugzilla.mozilla.org/show_bug.cgi?id=1857080#c10) might shed some light on this.